src/ci/notifier.rs

```diff
@@ -0,0 +1,10 @@
+pub mod github;
+pub mod gitlab;
+use crate::template;
+
+use anyhow::Result;
+
+pub trait Notifiable {
+    fn notify(&self, body: template::Template, patch: bool) -> Result<()>;
```

Comment:
```suggestion
    fn notify(&self, body: &template::Template, patch: bool) -> Result<()>;
```

should we need ownership of template::Template.
It can seem to be borrowed